.github/actions/setup-docker/action.yml

```diff
@@ -66,6 +66,32 @@ runs:
         echo "Docker binaries installed successfully"
         docker --version
 
+    - name: Install retry script
+      shell: bash
+      run: |
+        cat <<'EOF' | sudo tee /usr/local/bin/retry
+        #!/bin/bash
+        set -euo pipefail
+
+        max_attempts="${RETRY_MAX_ATTEMPTS:-5}"
+        delay="${RETRY_DELAY:-5}"
+        attempt=1
+
+        while true; do
+          if "$@"; then
+            exit 0
+          fi
+          if [[ $attempt -ge $max_attempts ]]; then
+            echo "Command failed after $max_attempts attempts: $*" >&2
+            exit 1
+          fi
+          echo "Attempt $attempt/$max_attempts failed, retrying in ${delay}s..." >&2
+          sleep $delay
+          attempt=$((attempt + 1))
+        done
+        EOF
+        sudo chmod +x /usr/local/bin/retry
```

Comment:
Should this be its own action maybe? I think we could also use that for some of the retries I added in https://github.com/firezone/firezone/pull/11449.